flink-ml-lib/src/main/java/org/apache/flink/ml/recommendation/swing/Swing.java

```diff
@@ -274,13 +272,29 @@ private static class ComputingSimilarItems extends AbstractStreamOperator<Row>
         private static final Character commaDelimiter = ',';
         private static final Character semicolonDelimiter = ';';
 
+        private final Random random;
+
+        private Map<Long, long[]> userAndPurchasedItems = new HashMap<>();
+        private Map<Long, List<Long>> itemAndPurchasers = new HashMap<>();
+
+        private ListState<Map<Long, long[]>> userAndPurchasedItemsState;
+        private ListState<Map<Long, List<Long>>> itemAndPurchasersState;
+
         private ComputingSimilarItems(
-                int k, int maxUserNumPerItem, int alpha1, int alpha2, double beta) {
+                int k,
+                int maxUserNumPerItem,
+                int maxUserBehavior,
+                int alpha1,
+                int alpha2,
+                double beta,
+                long seed) {
             this.k = k;
             this.maxUserNumPerItem = maxUserNumPerItem;
+            this.maxUserBehavior = maxUserBehavior;
             this.alpha1 = alpha1;
             this.alpha2 = alpha2;
             this.beta = beta;
+            this.random = new Random(seed);
```

Comment:
Is there a test case that uses a non-default seed? I would expect using different seeds would lead to different output but did not find it.